### PR TITLE
Discord enhancements

### PR DESCRIPTION
**Pull request by NiNiyas**
Discord

### CODE REVIEW

**Comment by louislam** [comment]
The "Test" button is not working.

image

**Comment by louislam** [comment]
And setting the URL is a bit strange. Maybe it can do it globally.

**Comment by NiNiyas** [comment]
> The "Test" button is not working.
> 
> image

So sorry for this. I think I fixed this. Could you please test it again?

File: server/notification.js [hunk 1]
```diff
@@ -86,38 +86,74 @@ class Notification {
...
+                                name: "Ping",
+                                value: heartbeatJSON["ping"] + "ms",
                             },
                         ],
                     }],
                 }
-                await axios.post(notification.discordWebhookUrl, data)
+                await axios.post(notification.discordWebhookUrl, discordupdata)
                 return okMsg;
+                }
```
Comment (Saibamen) [inline server/notification.js:156]: Indentation problem?

File: server/notification.js [hunk 2]
```diff
@@ -86,38 +86,74 @@ class Notification {
...
+                return okMsg;
+                    
+                } else if (heartbeatJSON["status"] == 1) {
+                    let discordupdata = {
+                    username: "Uptime Kuma",
                     embeds: [{
-                        title: "Uptime-Kuma Alert",
-                        color: alertColor,
+                        title: "✅ Your service " +  monitorJSON["name"] + " is up! ✅",
+                        color: 65280,
```
Comment (psykzz) [inline server/notification.js:132]: Would recommend replacing these with globals, like `color: colors.green`, same with the alert color.

File: server/notification.js [hunk 3]
```diff
@@ -86,38 +86,74 @@ class Notification {
...
+                    await axios.post(notification.discordWebhookUrl, discordtestdata)
                     return okMsg;
                 }
                 // If heartbeatJSON is not null, we go into the normal alerting loop.
                 if (heartbeatJSON["status"] == 0) {
-                    var alertColor = "16711680";
-                } else if (heartbeatJSON["status"] == 1) {
-                    var alertColor = "65280";
+                    let discorddowndata = {
+                    username: "Uptime Kuma",
```
Comment (Misly16) [inline server/notification.js:99]: Maybe a way use a custom name?

File: server/notification.js (hunk 2 above)
Comment (NiNiyas) [reply server/notification.js:132]: Hi! Isn't that for discordjs?. This is just a webhook payload. See here: https://leovoel.github.io/embed-visualizer/

File: server/notification.js (hunk 3 above)
Comment (NiNiyas) [reply server/notification.js:99]: Yeah, good idea. Any other suggestions?

**Review by Misly16 — APPROVED** [review]

**Pull request merged**